pkg/controller/servicebindingrequest/servicecontext.go

```diff
@@ -0,0 +1,146 @@
+package servicebindingrequest
+
+import (
+	"github.com/imdario/mergo"
+	"github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/annotations"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+
+	v1alpha1 "github.com/redhat-developer/service-binding-operator/pkg/apis/apps/v1alpha1"
+)
+
+// ServiceContext contains information related to a service.
+type ServiceContext struct {
+	// Object is the resource being used as reference.
+	Object *unstructured.Unstructured
+	// EnvVars contains the service's contributed environment variables.
+	EnvVars map[string]interface{}
+	// VolumeKeys contains the keys that should be mounted as volume from the binding secret.
+	VolumeKeys []string
+	// EnvVarPrefix indicates the prefix to use in environment variables.
+	EnvVarPrefix *string
+}
+
+// ServiceContextList is a list of ServiceContext values.
+type ServiceContextList []*ServiceContext
+
+// GetObjects returns a slice of service unstructured objects contained in the collection.
+func (sc ServiceContextList) GetObjects() []*unstructured.Unstructured {
+	var crs []*unstructured.Unstructured
+	for _, s := range sc {
+		crs = append(crs, s.Object)
+	}
+	return crs
+}
+
+// buildServiceContexts return a collection of ServiceContext values from the given service
+// selectors.
+//
+// TODO(isuttonl): implement tests
```

Comment:
they should be part of this PR

Reply:
Increase coverage in existing tests #471